doc/basics.md

```diff
@@ -168,6 +168,24 @@ And another one:
   cout << "number: " << v << endl;
 ```
 
+Minifying JSON strings without parsing
+----------------------
+
+In some cases, you may have valid JSON strings that you do not wish to parse but that you wish to minify. That is, you wish to remove all unnecessary spaces. We have a fast function for this purpose (`minify_string`). This function does not validate your content, and it does not parse it. Instead, it assumes that your string is valid UTF-8. It is much faster than parsing the string and re-serializing it in minified form. Usage is relatively simple. You must pass an input pointer with a length parameter, as well as an output pointer and an output length parameter (by reference). The output length parameter is not read, but written to. The output pointer should point to a valid memory region that is slightly overallocated (by `simdjson::SIMDJSON_PADDING`) compared to the original string length. The input pointer and input length are read, but not written to.
```

Comment:
I'm just going to point out that I decided to put this comment in as sort of "API futureproofing," but isn't actually true, if I recall correctly. Stage 1 runs against input without any padding. It's fine to leave it in, I'm just mentioning it because I'm still conflicted about my decision :)

Reply:
> I'm just going to point out that I decided to put the SIMDJSON_PADDING comment in as sort of "API futureproofing," but it isn't actually true, if I recall correctly. Stage 1 runs against input without any padding. It's fine to leave it in, I'm just mentioning it because I'm still conflicted about my decision :)

I do not understand. Can you elaborate ? Note that the `SIMDJSON_PADDING` here is not for input but rather for output, and it is 100% reasonable to have a bit more memory on the output side.

Reply:
Oh ... I missed that. No, I think that's super reasonable, so that we can blit simd-sized blocks onto the buffer.